src/commands/generate.ts

```diff
@@ -48,32 +53,73 @@ const generatePrompt = (templates: string[], load: any, options: OptionValues) =
 
-    if (fs.existsSync(path.resolve(projectName))) {
-      fs.rmSync(path.resolve(projectName), { recursive: true, force: true });
-    }
+    if (appConfirm == true) {
+      inquirer.prompt(APP).then(() => {
+        const appChoice = answers['app-choice'];
+        if (fs.existsSync(path.resolve(projectName))) {
+          fs.rmSync(path.resolve(projectName), { recursive: true, force: true });
+        }
+
+        load = !options.silent && loader('Generating template...');
+
+        exec(
+          `mkdir ${projectName} && cd ${projectName} && git init && git remote add -f origin https://github.com/warp-contracts/templates && git config core.sparseCheckout true && echo "${projectChoice}" >> .git/info/sparse-checkout && git config core.sparseCheckout false && echo "${appChoice}" >> .git/info/sparse-checkout && git pull origin main && git config core.sparseCheckout false && rm -rf .git`,
+          (error, stdout, stderr) => {
+            if (error) {
+              console.error(chalk.red.bold(`💣 [ERROR]:`), `Error while generating template: ${error.message} `);
+              load.stop();
+              return;
+            }
+            load.stop();
+            !options.silent &&
+              console.log(chalkGreen.bold(`🍭 [SUCCESS]:`), `Template generated in ${projectName} directory.`);
+          }
+        );
+      });
+    } else {
+      if (fs.existsSync(path.resolve(projectName))) {
+        fs.rmSync(path.resolve(projectName), { recursive: true, force: true });
+      }
+
+      load = !options.silent && loader('Generating template...');
 
-    load = !options.silent && loader('Generating template...');
+      exec(
+        `mkdir ${projectName} && cd ${projectName} && git init && git remote add -f origin https://github.com/warp-contracts/templates && git config core.sparseCheckout true && echo "${projectChoice}" >> .git/info/sparse-checkout  && git pull origin main && git config core.sparseCheckout false && rm -rf .git`,
+        (error, stdout, stderr) => {
+          if (error) {
+            console.error(chalk.red.bold(`💣 [ERROR]:`), `Error while generating template: ${error.message} `);
+            load.stop();
+            return;
+          }
 
-    exec(
-      `mkdir ${projectName} && cd ${projectName} && git init && git remote add -f origin https://github.com/warp-contracts/templates && git config core.sparseCheckout true && echo "${projectChoice}" >> .git/info/sparse-checkout && git pull origin main && git branch -m master main && git config core.sparseCheckout false`,
-      (error, stdout, stderr) => {
-        if (error) {
-          console.error(chalk.red.bold(`💣 [ERROR]:`), `Error while generating template: ${error.message} `);
           load.stop();
-          return;
+          !options.silent &&
+            console.log(chalkGreen.bold(`🍭 [SUCCESS]:`), `Template generated in ${projectName} directory.`);
         }
-        fs.renameSync(`${projectName}/${projectChoice}`, `${projectName}/contracts`);
```

Comment:
do we want to keep nested `typescript` folder? wouldnt it be better to just leave `contracts` directory?

Reply:
IMO it's more transparent as it is now. e.g. someone wants to add another folder with JS contract, then he/she don't  need to reorganize folder structure. But I can restore the changes if you want

Reply:
i dont think anyone will want js contracts along with ts contracts in one place so i'd leave just contracts folder